pkg/graveler/graveler.go

```diff
@@ -2145,7 +2151,7 @@ func (g *Graveler) Commit(ctx context.Context, repository *RepositoryRecord, bra
 			commit.Parents = CommitParents{branch.CommitID}
 		}
 
-		if !repository.ReadOnly {
+		if !repository.ReadOnly && !params.NoHooks {
```

Comment:
I don't like this, it breaks what pre-commit hooks guarantee!  For instance, [this blog post](https://lakefs.io/blog/lakefs-hooks/#clone-and-start-using-it-now) suggests using it to guarantee that commits have required metadata - and this by now an established pattern with some of our users.  Giving Lua hooks a pass from having metadata enforced is unsafe and unexpected.